src/metrics.rs

```diff
@@ -18,7 +18,7 @@ pub trait Producer {
 #[async_trait]
 pub trait Consumer {
     async fn consume(
-        self,
+        &self,
```

Comment:
any particular reason for `&self`?

Reply:
To write the unit test for `Print`, I had to keep a reference to the consumer(`Print`) after calling the `consume`.